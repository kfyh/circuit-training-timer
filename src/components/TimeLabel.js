import React from 'react'
import PropTypes from 'prop-types'

function TimeLabel ({ time }) {
  const displayTime = Math.ceil(Math.max(0, time / 1000))
  return (
    <div>
      Timer : {displayTime}
    </div>
  )
}

TimeLabel.propTypes = {
  time: PropTypes.number
}

export default TimeLabel
