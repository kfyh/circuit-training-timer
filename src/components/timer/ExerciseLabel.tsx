import React, { ReactElement } from 'react';
import { FlattenedExercise } from '../../types/circuits';

type ExerciseLabelProps = {
	exercise: FlattenedExercise;
	isResting: boolean;
};

export function ExerciseLabel(props: ExerciseLabelProps): ReactElement {
	const { exercise, isResting } = props;
	const excerciseLabel = exercise.name;

	return (
		<div>
			<p>
				{exercise.groupName} ({exercise.groupRepIndex}/{exercise.groupRepetitions})
			</p>
			{isResting ? (
				<p>Rest</p>
			) : (
				<p>
					{excerciseLabel} {exercise.count && exercise.count > 0 && `x ${exercise.count}`}
				</p>
			)}
		</div>
	);
}
