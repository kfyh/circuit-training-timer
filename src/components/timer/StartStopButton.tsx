import React, { ReactElement } from 'react';

type StartStopButtonProps = {
	isRunning: boolean;
	onClick: () => void;
}

export function StartStopButton({ isRunning, onClick }: StartStopButtonProps): ReactElement {
	const label = isRunning ? 'Stop' : 'Start';
	return <button onClick={onClick}>{label}</button>;
}
