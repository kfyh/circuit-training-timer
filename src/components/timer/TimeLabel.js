import React from 'react';
import PropTypes from 'prop-types';

export function TimeLabel({ time }) {
	const divStyle = {
		color: 'blue',
		fontSize: '250pt',
	};

	let displayTime = Math.ceil(Math.max(0, time / 1000));
	if (displayTime > 60) {
		displayTime = Math.ceil(time / 1000 / 60) + ' mins';
		divStyle.fontSize = '130pt';
	}

	return <div style={divStyle}>{displayTime}</div>;
}

TimeLabel.propTypes = {
	time: PropTypes.number,
};
