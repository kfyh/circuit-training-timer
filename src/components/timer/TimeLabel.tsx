import React, { ReactElement } from 'react';

type TimeLabelProps = {
	time: number;
	isResting: boolean;
};

export function TimeLabel({ time, isResting }: TimeLabelProps): ReactElement {
	const divStyle = {
		color: 'blue',
		fontSize: '250pt',
	};

	if (isResting) {
		divStyle.color = 'green';
	}

	const timeSeconds = Math.ceil(Math.max(0, time / 1000));
	let displayTime = timeSeconds.toString();
	if (timeSeconds > 60) {
		displayTime = Math.ceil(time / 1000 / 60) + ' mins';
		divStyle.fontSize = '130pt';
	}

	return <div style={divStyle}>{displayTime}</div>;
}
