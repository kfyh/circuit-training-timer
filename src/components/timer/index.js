export * from './StartStopButton';
export * from './ExerciseLabel';
export * from './TimeLabel';
