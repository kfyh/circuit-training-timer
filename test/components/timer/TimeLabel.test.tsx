import React from 'react';
import { shallow } from 'enzyme';
import { TimeLabel } from '../../../src/components/timer/TimeLabel';

describe('Given TimeLabel', () => {
	const testValues = [
		[1000, 1],
		[1499, 2],
		[0, 0],
		[-1, 0],
		[100000, 0],
	];

	test.each(testValues)('When render with time %i then show %i sec', (value, expected) => {
		const render = shallow(<TimeLabel time={value} />);
		expect(render).toMatchSnapshot();
	});
});
