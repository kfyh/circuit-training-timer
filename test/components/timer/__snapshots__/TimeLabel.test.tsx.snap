// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given TimeLabel When render with time -1 then show 0 sec 1`] = `ShallowWrapper {}`;

exports[`Given TimeLabel When render with time 0 then show 0 sec 1`] = `ShallowWrapper {}`;

exports[`Given TimeLabel When render with time 1000 then show 1 sec 1`] = `ShallowWrapper {}`;

exports[`Given TimeLabel When render with time 1499 then show 2 sec 1`] = `ShallowWrapper {}`;

exports[`Given TimeLabel When render with time 100000 then show 0 sec 1`] = `ShallowWrapper {}`;
